File: all-submissions/0871-keys-and-rooms/solution.java
class Solution {
    Set<Integer> seen = new HashSet<>();

    public boolean canVisitAllRooms(List<List<Integer>> rooms) {
        seen.add(0);
        dfs(0, rooms);
        if (seen.size() == rooms.size()) {
            return true;
        }
        return false;
    }

    public void dfs (int room_num, List<List<Integer>> rooms) {
        for (int room : rooms.get(room_num)) {
            if (!seen.contains(room)) {
                seen.add(room);
                dfs(room, rooms);
            }
        }
    }


}

File: all-submissions/0871-keys-and-rooms/solution.py
class Solution:
    def canVisitAllRooms(self, rooms: List[List[int]]) -> bool:
        n = len(rooms)
        visited = set()
        def dfs(room):
            if room not in visited:
                visited.add(room)
                for nr in rooms[room]:
                    dfs(nr)
        dfs(0)
        if len(visited) == n:
            return True
        return False
